spec/classes/application_spec.rb

```diff
@@ -140,61 +113,13 @@ class { 'katello::params':
         it { is_expected.to compile.with_all_deps }
 
         let(:katello_yaml_content) do
-          if facts[:operatingsystemmajrelease] == '7'
-            [
-              ':katello:',
-              '  :rest_client_timeout: 4000',
-              '  :content_types:',
-              '    :yum: true',
-              '    :file: true',
-              '    :deb: true',
-              '    :puppet: false',
-              '    :docker: true',
-              '    :ostree: false',
-              '  :candlepin:',
-              '    :url: https://localhost:23443/candlepin',
-              '    :oauth_key: "katello"',
-              '    :oauth_secret: "candlepin-secret"',
-              '    :ca_cert_file: /etc/pki/katello/certs/katello-default-ca.crt',
-              '  :candlepin_events:',
-              '    :ssl_cert_file: /etc/pki/katello/certs/java-client.crt',
-              '    :ssl_key_file: /etc/pki/katello/private/java-client.key',
-              '    :ssl_ca_file: /etc/pki/katello/certs/katello-default-ca.crt',
-              '  :agent:',
-              '    :broker_url: amqps://localhost:5671',
-              '    :event_queue_name: katello.agent',
-              '  :katello_applicability: true',
-            ]
-          else
-            [
-              ':katello:',
-              '  :rest_client_timeout: 4000',
-              '  :content_types:',
-              '    :yum: true',
-              '    :file: true',
-              '    :deb: true',
-              '    :puppet: false',
-              '    :docker: true',
-              '    :ostree: false',
-              '  :candlepin:',
-              '    :url: https://localhost:23443/candlepin',
-              '    :oauth_key: "katello"',
-              '    :oauth_secret: "candlepin-secret"',
-              '    :ca_cert_file: /etc/pki/katello/certs/katello-default-ca.crt',
-              '  :candlepin_events:',
-              '    :ssl_cert_file: /etc/pki/katello/certs/java-client.crt',
-              '    :ssl_key_file: /etc/pki/katello/private/java-client.key',
-              '    :ssl_ca_file: /etc/pki/katello/certs/katello-default-ca.crt',
-              '  :agent:',
-              '    :broker_url: amqps://localhost:5671',
-              '    :event_queue_name: katello.agent',
-              '  :katello_applicability: true',
-            ]
-          end
+          [
+            '  :rest_client_timeout: 4000',
+          ]
         end
 
         it 'should generate correct katello.yaml' do
-          verify_exact_contents(catalogue, '/etc/foreman/plugins/katello.yaml', katello_yaml_content)
+          verify_contents(catalogue, '/etc/foreman/plugins/katello.yaml', katello_yaml_content)
```

Comment:
Thinking out loud: does it make sense to shorten this and drop the `let` block?
```suggestion
          verify_contents(catalogue, '/etc/foreman/plugins/katello.yaml', ['  :rest_client_timeout: 4000'])
```

On the other hand, it keeps the tests consistent since they all follow the same pattern.

Reply:
Yea, I both liked the pattern nature of the tests and the fact that it 'looks' like yaml which I find as a person writing or reading the tests easier to translate to a deployment.